core/src/service/path_service.rs

```diff
@@ -148,6 +149,27 @@ pub fn get_all_paths(config: &Config, filter: Option<Filter>) -> Result<Vec<Stri
     Ok(paths)
 }
 
+pub fn get_path_by_id(config: &Config, id: Uuid) -> Result<String, CoreError> {
+    let mut current_id = id;
+    let mut current_metadata = file_metadata_repo::get(config, current_id)?;
+    let mut path = String::from("");
+
+    while current_metadata.parent != current_id {
+        let name = file_encryption_service::get_name(&config, &current_metadata)?;
+        path = format!("{}/{}", name, path);
+        current_id = current_metadata.parent;
+        current_metadata = file_metadata_repo::get(config, current_id)?
+    }
+
+    {
+        let name = file_encryption_service::get_name(&config, &current_metadata)?;
+        path = format!("{}/{}", name, path);
+    }
+    // Remove the last forward slash.
+    path.pop();
```

Comment:
If you're only using `"{}/{}"` where does the trailing forward slash come from?

Also I think if the last element in a path is a folder, we want that trailing slash.

Reply:
path is always empty at first, so there is always a trailing '/'.  I could, however, just not do this if it is a folder.